core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/stack/StackRequestValidator.java

```diff
@@ -69,5 +85,27 @@ private void validateTemplates(StackRequest stackRequest, ValidationResultBuilde
                 .reduce(ValidationResult::merge)
                 .ifPresent(resultBuilder::merge);
     }
+
+    private void validateSharedService(StackRequest stackRequest, ValidationResultBuilder validationBuilder) {
+        if (stackRequest.getClusterToAttach() != null) {
+            boolean hasInAppropriateStatus = false;
+            Optional<Stack> stack = Optional.ofNullable(stackRepository.findOne(stackRequest.getClusterToAttach()));
+            if (stack.isPresent() && !AVAILABLE.equals(stack.get().getStatus())) {
+                hasInAppropriateStatus = true;
+            }
+            if (!hasInAppropriateStatus) {
+                for (Cluster cluster : clusterRepository.findAllByStackId(stackRequest.getClusterToAttach())) {
```

Comment:
`hasInAppropriateStatus = clusterRepository.findAllByStackId(stackRequest.getClusterToAttach()).stream().anyMatch(c -> c != null && !AVAILABLE.equals(c())` instead of foreach and break.